### PR TITLE
feat: markdown rendering of messages and more actions

### PR DESCRIPTION
**Pull request by bm424**
**Merging to feature branch**
* adds markdown rendering and styles matching the chat
* adds sticky scrolling (copied from chat)
* adds two new action buttons: "reset" and "retry"
See sourcery-ai/core#2876

### CODE REVIEW

File: src/troubleshooting.ts [hunk 1]
```diff
@@ -2,21 +2,47 @@
...
 } from "vscode";
 import { randomBytes } from "crypto";
+import { marked } from "marked";
+import sanitizeHtml from "sanitize-html";
+
+// NOTE: marked code block formatting relies on the `marked.use` block in `chat.ts`
 
-type TroubleshootingResult = any;
+type TroubleshootingResult = {
+  type: "error" | "assistance" | "feedback" | "user" | string;
```
Comment (Hellebore) [inline src/troubleshooting.ts:21]: having string here seems a little odd

File: src/troubleshooting.ts [hunk 2]
```diff
@@ -59,6 +93,17 @@ export class TroubleshootingProvider implements WebviewViewProvider {
...
+    if (window.activeColorTheme.kind === ColorThemeKind.Light) {
+      hljsUri = this._view.webview.asWebviewUri(
+        Uri.joinPath(this._extensionUri, "media", "github.min.css")
+      );
+    } else {
+      hljsUri = this._view.webview.asWebviewUri(
+        Uri.joinPath(this._extensionUri, "media", "github-dark.min.css")
+      );
+    }
+
```
Comment (Hellebore) [inline src/troubleshooting.ts:106]: it would be nice to share this rendering and markdown code with the chat

**Review by Hellebore — APPROVED** [review]
Cool - I guess we will need to refactor once we settle on a final UI

File: src/troubleshooting.ts (hunk 1 above)
Comment (bm424) [reply src/troubleshooting.ts:21]: Removed in the next PR, I was kind of thinking "this is a free entry type but these are the expected ones"

File: src/troubleshooting.ts (hunk 2 above)
Comment (bm424) [reply src/troubleshooting.ts:106]: Agree

**Pull request merged**